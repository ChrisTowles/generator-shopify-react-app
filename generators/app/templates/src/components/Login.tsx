import { Button, DisplayText, FormLayout, TextField } from "@shopify/polaris";
import * as React from "react";

import * as styles from "./login.scss";

interface ILoginProps {
    disableInstall: boolean;
    handleSubmit: () => void;
    handleStoreChanged: (value: string, id: string) => void;
    errorMessage: string | undefined;
    installMessage: string;
    shop: string;
}

export function Login(props: ILoginProps) {
    return (
        <main role="main" className={styles.main}>
            <header className={styles.header}>
                <DisplayText size="extraLarge">Shopify App — Installation</DisplayText>
            </header>
            <div className={styles.form}>
                <FormLayout>
                    <TextField
                        error={props.errorMessage}
                        id="shop"
                        label="Please enter the “myshopify” domain of your store"
                        name="shop"
                        onChange={props.handleStoreChanged}
                        placeholder="example.myshopify.com"
                        value={props.shop}
                    />
                    <Button
                        primary
                        fullWidth
                        onClick={props.handleSubmit}
                        size="large"
                        disabled={props.disableInstall}>{props.installMessage}</Button>
                </FormLayout>
            </div>
        </main>
    );
}
