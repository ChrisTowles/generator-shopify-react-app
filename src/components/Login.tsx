import * as React from "react";

class LoginProps {
    handleSubmit: React.EventHandler<React.FormEvent<HTMLFormElement>>;
    handleStoreChanged: React.EventHandler<React.ChangeEvent<HTMLInputElement>>;
    errorMessage: string | null;
    shop: string;
}

export function Login(props: LoginProps) {
    const error = props.errorMessage ? <div className="alert error"><dl><dt>Error Alert</dt><dd>{props.errorMessage}</dd></dl></div> : null;
    return (
        <main className="container" role="main">
            <header>
                <h1>Shopify App — Installation</h1>
                <p className="subhead">
                    <label htmlFor="shop">Please enter the “myshopify” domain of your store</label>
                </p>
            </header>

            <div className="container__form">
                {error}
                <form method="GET" action="login" onSubmit={props.handleSubmit}>
                    <input type="text" name="shop" placeholder="example.myshopify.com" defaultValue={props.shop} onChange={props.handleStoreChanged} />
                    <button type="submit">Install</button>
                </form>
            </div>
        </main>
    );
}