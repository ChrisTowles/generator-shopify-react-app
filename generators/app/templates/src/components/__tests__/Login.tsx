import * as React from "react";
import * as renderer from "react-test-renderer";
import { Login } from "../Login";

test("Renders correctly with no error or shop", () => {
  const login = <Login
    handleStoreChanged={() => { return; }}
    handleSubmit={() => { return; }}
    errorMessage={null}
    shop={""} />;

  const component = renderer.create(login);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Renders correctly with shop", () => {
  const login = <Login
    handleStoreChanged={() => { return; }}
    handleSubmit={() => { return; }}
    errorMessage={null}
    shop={"example.myshopify.com"} />;

  const component = renderer.create(login);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Renders correctly with error", () => {
  const login = <Login
    handleStoreChanged={() => { return; }}
    handleSubmit={() => { return; }}
    errorMessage={"Store domain name is wrong"}
    shop={"example.myshopify.comx"} />;

  const component = renderer.create(login);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
