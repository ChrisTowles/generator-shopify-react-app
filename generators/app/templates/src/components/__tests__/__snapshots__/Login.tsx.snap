// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with error 1`] = `
<main
  className="container"
  role="main"
>
  <header>
    <h1>
      Shopify App — Installation
    </h1>
    <p
      className="subhead"
    >
      <label
        htmlFor="shop"
      >
        Please enter the “myshopify” domain of your store
      </label>
    </p>
  </header>
  <div
    className="container__form"
  >
    <div
      className="alert error"
    >
      <dl>
        <dt>
          Error Alert
        </dt>
        <dd>
          Store domain name is wrong
        </dd>
      </dl>
    </div>
    <form
      action="login"
      method="GET"
      onSubmit={[Function]}
    >
      <input
        defaultValue="example.myshopify.comx"
        name="shop"
        onChange={[Function]}
        placeholder="example.myshopify.com"
        type="text"
      />
      <button
        type="submit"
      >
        Install
      </button>
    </form>
  </div>
</main>
`;

exports[`Renders correctly with no error or shop 1`] = `
<main
  className="container"
  role="main"
>
  <header>
    <h1>
      Shopify App — Installation
    </h1>
    <p
      className="subhead"
    >
      <label
        htmlFor="shop"
      >
        Please enter the “myshopify” domain of your store
      </label>
    </p>
  </header>
  <div
    className="container__form"
  >
    <form
      action="login"
      method="GET"
      onSubmit={[Function]}
    >
      <input
        defaultValue=""
        name="shop"
        onChange={[Function]}
        placeholder="example.myshopify.com"
        type="text"
      />
      <button
        type="submit"
      >
        Install
      </button>
    </form>
  </div>
</main>
`;

exports[`Renders correctly with shop 1`] = `
<main
  className="container"
  role="main"
>
  <header>
    <h1>
      Shopify App — Installation
    </h1>
    <p
      className="subhead"
    >
      <label
        htmlFor="shop"
      >
        Please enter the “myshopify” domain of your store
      </label>
    </p>
  </header>
  <div
    className="container__form"
  >
    <form
      action="login"
      method="GET"
      onSubmit={[Function]}
    >
      <input
        defaultValue="example.myshopify.com"
        name="shop"
        onChange={[Function]}
        placeholder="example.myshopify.com"
        type="text"
      />
      <button
        type="submit"
      >
        Install
      </button>
    </form>
  </div>
</main>
`;
